app/code/Magento/SalesRule/Model/Coupon/UpdateCouponUsages.php

```diff
@@ -0,0 +1,153 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\SalesRule\Model\Coupon;
+
+use Magento\Sales\Model\Order;
+use Magento\SalesRule\Model\Coupon;
+use Magento\SalesRule\Model\ResourceModel\Coupon\Usage;
+use Magento\SalesRule\Model\Rule\CustomerFactory;
+use Magento\SalesRule\Model\RuleFactory;
+
+/**
+ * Updates the coupon usages.
+ */
+class UpdateCouponUsages
+{
+    /**
+     * @var RuleFactory
+     */
+    private $ruleFactory;
+
+    /**
+     * @var RuleFactory
+     */
+    private $ruleCustomerFactory;
+
+    /**
+     * @var Coupon
+     */
+    private $coupon;
+
+    /**
+     * @var Usage
+     */
+    private $couponUsage;
+
+    /**
+     * @param RuleFactory $ruleFactory
+     * @param CustomerFactory $ruleCustomerFactory
+     * @param Coupon $coupon
+     * @param Usage $couponUsage
+     */
+    public function __construct(
+        RuleFactory $ruleFactory,
+        CustomerFactory $ruleCustomerFactory,
+        Coupon $coupon,
+        Usage $couponUsage
+    ) {
+        $this->ruleFactory = $ruleFactory;
+        $this->ruleCustomerFactory = $ruleCustomerFactory;
+        $this->coupon = $coupon;
+        $this->couponUsage = $couponUsage;
+    }
+
+    /**
+     * Executes the current command.
+     *
+     * @param Order $subject
+     * @param bool $increment
+     * @return Order
+     */
+    public function execute(Order $subject, bool $increment)
```

Comment:
method should have explicit return type
```suggestion
    public function execute(Order $subject, bool $increment): Order
```